src/Microsoft.OData.Client/Materialization/EntryValueMaterializationPolicy.cs

```diff
@@ -611,7 +611,7 @@ private void MaterializeResolvedEntry(MaterializerEntry entry, bool includeLinks
                     }
 
                     // includeLinks is for Navigation property, so we should handle complex property when includeLinks equals false;
-                    if (!includeLinks && (prop.IsEntityCollection || prop.EntityCollectionItemType != null))
+                    if (!includeLinks && prop.EdmProperty is IEdmNavigationProperty)

```

Comment:
The intention of the original code was to check:
-  If includeLinks is not true, and
-  If the property is not a collection of complex type.

That code didn't make sense since it ignored single-value navigation property.

In-short `includeLinks` should only apply to navigation properties (single-value and collections) and ignore all complex types.